### PR TITLE
DPC-518 log lookback stats, add jobid, batchid and patient mbi hash to mdc

### PR DESCRIPTION
**Pull request by whuang85**
### Proposed Changes

- Add jobID, batchID and patient MBI Hash to MDC for logs
- Log lookBack stats for each patient

### Security Implications

Make sure MBI Hash is not PII

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

```
{"level":"INFO","logger":"gov.cms.dpc.aggregation.engine.AggregationEngine","thread":"RxComputationThreadPool-1","message":"Processing job, exporting to: /tmp.","mdc":{"jobID":"c925679a-ff7f-4a33-862c-14e46388b3d5","batchID":"1de6b574-5bd5-4b57-8a2f-a5a3597d1206"},"timestamp":"2020-07-09T14:44:35.087+0000"}
{"level":"INFO","logger":"gov.cms.dpc.aggregation.service.LookBackService","thread":"RxComputationThreadPool-1","message":"LookBack stats eobWithinLookBackLimit true, eobContainsProvider true, eobRelatedToOrganization true, eobMonthsDifference 2, hasClaim true","mdc":{"batchID":"1de6b574-5bd5-4b57-8a2f-a5a3597d1206","jobID":"c925679a-ff7f-4a33-862c-14e46388b3d5","patientID":"d35350fce12f555089f938c0323a13122622123038e8af057a4191fd450c2b90"},"timestamp":"2020-07-09T14:44:35.722+0000"}
{"level":"INFO","logger":"gov.cms.dpc.aggregation.service.LookBackService","thread":"RxComputationThreadPool-1","message":"LookBack stats eobWithinLookBackLimit true, eobContainsProvider true, eobRelatedToOrganization true, eobMonthsDifference 6, hasClaim true","mdc":{"jobID":"c925679a-ff7f-4a33-862c-14e46388b3d5","batchID":"1de6b574-5bd5-4b57-8a2f-a5a3597d1206","patientID":"41af07535e0a66226cf2f0e6c551c0a15bd49192fc055aa5cd2e63f31f90a419"},"timestamp":"2020-07-09T14:44:36.311+0000"}
{"level":"INFO","logger":"gov.cms.dpc.aggregation.service.LookBackService","thread":"RxComputationThreadPool-1","message":"eob BillingPeriod or job providerID or job organizationID or eob OrganizationID are null","mdc":{"jobID":"c925679a-ff7f-4a33-862c-14e46388b3d5","batchID":"1de6b574-5bd5-4b57-8a2f-a5a3597d1206","patientID":"e411277fd31da392eaa9a45df53b0c429e365626182f50d9f35810d77f0e2756"},"timestamp":"2020-07-09T14:44:36.616+0000"}
{"level":"INFO","logger":"gov.cms.dpc.aggregation.service.LookBackService","thread":"RxComputationThreadPool-1","message":"LookBack stats eobWithinLookBackLimit true, eobContainsProvider true, eobRelatedToOrganization true, eobMonthsDifference 3, hasClaim true","mdc":{"jobID":"c925679a-ff7f-4a33-862c-14e46388b3d5","batchID":"1de6b574-5bd5-4b57-8a2f-a5a3597d1206","patientID":"e411277fd31da392eaa9a45df53b0c429e365626182f50d9f35810d77f0e2756"},"timestamp":"2020-07-09T14:44:36.621+0000"}
{"exception":"java.lang.NullPointerException: null\n\tat java.base/java.util.Objects.requireNonNull(Objects.java:221)\n\tat java.base/java.util.ImmutableCollections$MapN.containsValue(ImmutableCollections.java:847)\n\tat gov.cms.dpc.bluebutton.client.MockBlueButtonClient.loadResource(MockBlueButtonClient.java:156)\n\tat gov.cms.dpc.bluebutton.client.MockBlueButtonClient.loadBundle(MockBlueButtonClient.java:139)\n\tat gov.cms.dpc.bluebutton.client.MockBlueButtonClient.requestPatientFromServerByMbi(MockBlueButtonClient.java:63)\n\tat gov.cms.dpc.aggregation.engine.ResourceFetcher.fetchPatient(ResourceFetcher.java:151)\n\tat gov.cms.dpc.aggregation.engine.ResourceFetcher.fetchFirst(ResourceFetcher.java:128)\n\tat gov.cms.dpc.aggregation.engine.ResourceFetcher.lambda$fetchResources$0(ResourceFetcher.java:72)\n\tat io.reactivex.internal.operators.flowable.FlowableFromCallable.subscribeActual(FlowableFromCallable.java:39)\n\tat io.reactivex.Flowable.subscribe(Flowable.java:14935)\n\tat io.reactivex.internal.operators.flowable.FlowableOnErrorNext.subscribeActual(FlowableOnErrorNext.java:40)\n\tat io.reactivex.Flowable.subscribe(Flowable.java:14935)\n\tat io.reactivex.internal.operators.flowable.FlowableFlatMap.subscribeActual(FlowableFlatMap.java:53)\n\tat io.reactivex.Flowable.subscribe(Flowable.java:14935)\n\tat io.reactivex.internal.operators.flowable.FlowableFilter.subscribeActual(FlowableFilter.java:37)\n\tat io.reactivex.Flowable.subscribe(Flowable.java:14935)\n\tat io.reactivex.internal.operators.flowable.FlowableAnySingle.subscribeActual(FlowableAnySingle.java:37)\n\tat io.reactivex.Single.subscribe(Single.java:3666)\n\tat io.reactivex.internal.operators.single.SingleOnErrorReturn.subscribeActual(SingleOnErrorReturn.java:38)\n\tat io.reactivex.Single.subscribe(Single.java:3666)\n\tat io.reactivex.Single.blockingGet(Single.java:2869)\n\tat gov.cms.dpc.aggregation.engine.AggregationEngine.isValidLookBack(AggregationEngine.java:208)\n\tat gov.cms.dpc.aggregation.engine.AggregationEngine.processPatient(AggregationEngine.java:189)\n\tat gov.cms.dpc.aggregation.engine.AggregationEngine.processJobBatch(AggregationEngine.java:163)\n\tat io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)\n\tat io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:62)\n\tat io.reactivex.internal.operators.observable.ObservableFilter$FilterObserver.onNext(ObservableFilter.java:52)\n\tat io.reactivex.internal.operators.observable.ObservableRetryPredicate$RepeatObserver.onNext(ObservableRetryPredicate.java:69)\n\tat io.reactivex.internal.operators.observable.ObservableDoOnEach$DoOnEachObserver.onNext(ObservableDoOnEach.java:101)\n\tat io.reactivex.internal.operators.observable.ObservableDoOnEach$DoOnEachObserver.onNext(ObservableDoOnEach.java:101)\n\tat io.reactivex.internal.util.HalfSerializer.onNext(HalfSerializer.java:107)\n\tat io.reactivex.internal.operators.observable.ObservableRepeatWhen$RepeatWhenObserver.onNext(ObservableRepeatWhen.java:100)\n\tat io.reactivex.internal.observers.DeferredScalarDisposable.complete(DeferredScalarDisposable.java:82)\n\tat io.reactivex.internal.operators.observable.ObservableFromCallable.subscribeActual(ObservableFromCallable.java:53)\n\tat io.reactivex.Observable.subscribe(Observable.java:12284)\n\tat io.reactivex.internal.operators.observable.ObservableRepeatWhen$RepeatWhenObserver.subscribeNext(ObservableRepeatWhen.java:151)\n\tat io.reactivex.internal.operators.observable.ObservableRepeatWhen$RepeatWhenObserver.innerNext(ObservableRepeatWhen.java:128)\n\tat io.reactivex.internal.operators.observable.ObservableRepeatWhen$RepeatWhenObserver$InnerRepeatObserver.onNext(ObservableRepeatWhen.java:168)\n\tat io.reactivex.observers.SerializedObserver.onNext(SerializedObserver.java:111)\n\tat io.reactivex.internal.operators.observable.ObservableDelay$DelayObserver$OnNext.run(ObservableDelay.java:114)\n\tat io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)\n\tat io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)\n\tat java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\n\tat java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.base/java.lang.Thread.run(Thread.java:834)\n","level":"ERROR","logger":"gov.cms.dpc.aggregation.engine.ResourceFetcher","thread":"RxComputationThreadPool-1","message":"Turning error into OperationOutcome.","mdc":{"jobID":"c925679a-ff7f-4a33-862c-14e46388b3d5","batchID":"1de6b574-5bd5-4b57-8a2f-a5a3597d1206","patientID":"e411277fd31da392eaa9a45df53b0c429e365626182f50d9f35810d77f0e2756"},"timestamp":"2020-07-09T14:44:36.892+0000"}
{"level":"INFO","logger":"gov.cms.dpc.aggregation.engine.AggregationEngine","thread":"RxComputationThreadPool-1","message":"COMPLETED job","mdc":{"jobID":"c925679a-ff7f-4a33-862c-14e46388b3d5","batchID":"1de6b574-5bd5-4b57-8a2f-a5a3597d1206"},"timestamp":"2020-07-09T14:44:36.895+0000"}
{"level":"INFO","logger":"gov.cms.dpc.aggregation.engine.AggregationEngine","thread":"RxComputationThreadPool-1","message":"Processing job, exporting to: /tmp.","mdc":{"batchID":"1700017d-05fe-4eaf-9d1c-403feedf11e7","jobID":"f84a5c0e-587e-45fb-83d0-8529ef9bdb3d"},"timestamp":"2020-07-09T14:44:39.711+0000"}
{"level":"INFO","logger":"gov.cms.dpc.aggregation.engine.AggregationEngine","thread":"RxComputationThreadPool-1","message":"COMPLETED job","mdc":{"batchID":"1700017d-05fe-4eaf-9d1c-403feedf11e7","jobID":"f84a5c0e-587e-45fb-83d0-8529ef9bdb3d"},"timestamp":"2020-07-09T14:44:39.711+0000"}
{"level":"INFO","logger":"gov.cms.dpc.aggregation.engine.AggregationEngine","thread":"RxComputationThreadPool-1","message":"Processing job, exporting to: /tmp.","mdc":{"batchID":"4f871117-24a5-4cbb-8698-d9c5fe71ca78","jobID":"73bf6362-0925-482e-a498-c8ed73809d06"},"timestamp":"2020-07-09T14:44:42.286+0000"}
{"level":"INFO","logger":"gov.cms.dpc.aggregation.engine.AggregationEngine","thread":"RxComputationThreadPool-1","message":"COMPLETED job","mdc":{"batchID":"4f871117-24a5-4cbb-8698-d9c5fe71ca78","jobID":"73bf6362-0925-482e-a498-c8ed73809d06"},"timestamp":"2020-07-09T14:44:42.287+0000"}
{"level":"INFO","logger":"gov.cms.dpc.aggregation.engine.AggregationEngine","thread":"RxComputationThreadPool-1","message":"Processing job, exporting to: /tmp.","mdc":{"batchID":"d1d7e45a-7bee-40ae-bb29-625906acdf73","jobID":"c0439095-1ba3-4971-b074-5dc23bf491c8"},"timestamp":"2020-07-09T14:44:45.895+0000"}
{"exception":"java.lang.NullPointerException: null\n\tat java.base/java.util.Objects.requireNonNull(Objects.java:221)\n\tat java.base/java.util.ImmutableCollections$MapN.containsValue(ImmutableCollections.java:847)\n\tat gov.cms.dpc.bluebutton.client.MockBlueButtonClient.loadResource(MockBlueButtonClient.java:156)\n\tat gov.cms.dpc.bluebutton.client.MockBlueButtonClient.loadBundle(MockBlueButtonClient.java:139)\n\tat gov.cms.dpc.bluebutton.client.MockBlueButtonClient.requestPatientFromServerByMbi(MockBlueButtonClient.java:63)\n\tat gov.cms.dpc.aggregation.engine.ResourceFetcher.fetchPatient(ResourceFetcher.java:151)\n\tat gov.cms.dpc.aggregation.engine.ResourceFetcher.fetchFirst(ResourceFetcher.java:128)\n\tat gov.cms.dpc.aggregation.engine.ResourceFetcher.lambda$fetchResources$0(ResourceFetcher.java:72)\n\tat io.reactivex.internal.operators.flowable.FlowableFromCallable.subscribeActual(FlowableFromCallable.java:39)\n\tat io.reactivex.Flowable.subscribe(Flowable.java:14935)\n\tat io.reactivex.internal.operators.flowable.FlowableOnErrorNext.subscribeActual(FlowableOnErrorNext.java:40)\n\tat io.reactivex.Flowable.subscribe(Flowable.java:14935)\n\tat io.reactivex.internal.operators.flowable.FlowableFlatMap.subscribeActual(FlowableFlatMap.java:53)\n\tat io.reactivex.Flowable.subscribe(Flowable.java:14935)\n\tat io.reactivex.internal.operators.flowable.FlowableFilter.subscribeActual(FlowableFilter.java:37)\n\tat io.reactivex.Flowable.subscribe(Flowable.java:14935)\n\tat io.reactivex.internal.operators.flowable.FlowableAnySingle.subscribeActual(FlowableAnySingle.java:37)\n\tat io.reactivex.Single.subscribe(Single.java:3666)\n\tat io.reactivex.internal.operators.single.SingleOnErrorReturn.subscribeActual(SingleOnErrorReturn.java:38)\n\tat io.reactivex.Single.subscribe(Single.java:3666)\n\tat io.reactivex.Single.blockingGet(Single.java:2869)\n\tat gov.cms.dpc.aggregation.engine.AggregationEngine.isValidLookBack(AggregationEngine.java:208)\n\tat gov.cms.dpc.aggregation.engine.AggregationEngine.processPatient(AggregationEngine.java:189)\n\tat gov.cms.dpc.aggregation.engine.AggregationEngine.processJobBatch(AggregationEngine.java:163)\n\tat io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)\n\tat io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:62)\n\tat io.reactivex.internal.operators.observable.ObservableFilter$FilterObserver.onNext(ObservableFilter.java:52)\n\tat io.reactivex.internal.operators.observable.ObservableRetryPredicate$RepeatObserver.onNext(ObservableRetryPredicate.java:69)\n\tat io.reactivex.internal.operators.observable.ObservableDoOnEach$DoOnEachObserver.onNext(ObservableDoOnEach.java:101)\n\tat io.reactivex.internal.operators.observable.ObservableDoOnEach$DoOnEachObserver.onNext(ObservableDoOnEach.java:101)\n\tat io.reactivex.internal.util.HalfSerializer.onNext(HalfSerializer.java:107)\n\tat io.reactivex.internal.operators.observable.ObservableRepeatWhen$RepeatWhenObserver.onNext(ObservableRepeatWhen.java:100)\n\tat io.reactivex.internal.observers.DeferredScalarDisposable.complete(DeferredScalarDisposable.java:82)\n\tat io.reactivex.internal.operators.observable.ObservableFromCallable.subscribeActual(ObservableFromCallable.java:53)\n\tat io.reactivex.Observable.subscribe(Observable.java:12284)\n\tat io.reactivex.internal.operators.observable.ObservableRepeatWhen$RepeatWhenObserver.subscribeNext(ObservableRepeatWhen.java:151)\n\tat io.reactivex.internal.operators.observable.ObservableRepeatWhen$RepeatWhenObserver.innerNext(ObservableRepeatWhen.java:128)\n\tat io.reactivex.internal.operators.observable.ObservableRepeatWhen$RepeatWhenObserver$InnerRepeatObserver.onNext(ObservableRepeatWhen.java:168)\n\tat io.reactivex.observers.SerializedObserver.onNext(SerializedObserver.java:111)\n\tat io.reactivex.internal.operators.observable.ObservableDelay$DelayObserver$OnNext.run(ObservableDelay.java:114)\n\tat io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)\n\tat io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)\n\tat java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\n\tat java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.base/java.lang.Thread.run(Thread.java:834)\n","level":"ERROR","logger":"gov.cms.dpc.aggregation.engine.ResourceFetcher","thread":"RxComputationThreadPool-1","message":"Turning error into OperationOutcome.","mdc":{"batchID":"d1d7e45a-7bee-40ae-bb29-625906acdf73","jobID":"c0439095-1ba3-4971-b074-5dc23bf491c8"},"timestamp":"2020-07-09T14:44:45.904+0000"}
{"level":"INFO","logger":"gov.cms.dpc.aggregation.engine.AggregationEngine","thread":"RxComputationThreadPool-1","message":"COMPLETED job","mdc":{"batchID":"d1d7e45a-7bee-40ae-bb29-625906acdf73","jobID":"c0439095-1ba3-4971-b074-5dc23bf491c8"},"timestamp":"2020-07-09T14:44:45.904+0000"}
```

### CODE REVIEW

**Comment by embh** [comment]
BFD RFC indicating that the purpose of the hashed MBIs is to prevent PII in logs: https://github.com/CMSgov/beneficiary-fhir-data/blob/master/rfcs/0005-mbi-search.md#hash-algorithm

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/AggregationEngine.java [hunk 1]
```diff
@@ -159,19 +163,21 @@ protected void processJobBatch(JobQueueBatch job) {
...
                 this.queue.completeBatch(job, aggregatorID);
             } else {
-                logger.info("PAUSED job {} batch {}", job.getJobID(), job.getBatchID());
+                logger.info("PAUSED job");
                 this.queue.pauseBatch(job, aggregatorID);
             }
         } catch (Exception error) {
             try {
-                logger.error("FAILED job {} batch {}", job.getJobID(), job.getBatchID(), error);
+                logger.error("FAILED job", error);
```
Comment (dhgreene) [inline dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/AggregationEngine.java:180]: Could you include some example log output in the PR description?  I'd like to be certain all these log statements can be correlated.

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/service/LookBackServiceImpl.java [hunk 2]
```diff
@@ -57,12 +61,24 @@ public boolean hasClaimWithin(ExplanationOfBenefit explanationOfBenefit, UUID or
...
-                && organizationID.isPresent() && eobOrganizationID.isPresent()
-                && getMonthsDifference(billingPeriod.get(), operationsConfig.getLookBackDate()) < withinMonth
-                && eobProviderNPIs.contains(providerID.get())
-                && organizationID.get().equals(eobOrganizationID.get());
+        if (billingPeriod.isEmpty() || providerID.isEmpty() || organizationID.isEmpty() || eobOrganizationID.isEmpty()) {
+            LOGGER.info("eob BillingPeriod or job providerID or job organizationID or eob OrganizationID are null");
+            return false;
+        }
+
+        long lookBackMonthsDifference = getMonthsDifference(billingPeriod.get(), operationsConfig.getLookBackDate());
```
[the remark this answers is not in the document]
Comment (whuang85) [reply dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/service/LookBackServiceImpl.java:69]: Checking the present/absence of value here: https://github.com/CMSgov/dpc-app/pull/897/files/539e7ab69c26846201112275e96df2cbd5d75a95#diff-928f42d0deeb5304fb975977200c88e3R64

Not sure what code climate is complaining about..

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/AggregationEngine.java (hunk 1 above)
Comment (whuang85) [reply dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/AggregationEngine.java:180]: > Could you include some example log output in the PR description? I'd like to be certain all these log statements can be correlated.

done

**Pull request merged**